From: Maciek Borzecki <maciek.borzecki@gmail.com>
Date: Sat, 24 Sep 2016 14:17:30 +0200
Subject: build-sys: get rid of move-to-rootlibdir

Replace move-to-rootlibdir calls in post-install hooks with explicitly
used ${rootlibdir} where needed.

Signed-off-by: Maciek Borzecki <maciek.borzecki@gmail.com>
---
 Makefile.am | 33 ++-------------------------------
 1 file changed, 2 insertions(+), 31 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index c668187..665abc6 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -251,16 +251,6 @@ AM_CFLAGS = $(OUR_CFLAGS)
 AM_LDFLAGS = $(OUR_LDFLAGS)
 
 # ------------------------------------------------------------------------------
-define move-to-rootlibdir
-	if test "$(libdir)" != "$(rootlibdir)"; then \
-		$(MKDIR_P) $(DESTDIR)$(rootlibdir) && \
-		so_img_name=$$(readlink $(DESTDIR)$(libdir)/$$libname) && \
-		rm -f $(DESTDIR)$(libdir)/$$libname && \
-		$(LN_S) --relative -f $(DESTDIR)$(rootlibdir)/$$so_img_name $(DESTDIR)$(libdir)/$$libname && \
-		mv $(DESTDIR)$(libdir)/$$libname.* $(DESTDIR)$(rootlibdir); \
-	fi
-endef
-
 INSTALL_DIRS =
 
 SHUTDOWN_TARGET_WANTS =
@@ -3258,15 +3248,6 @@ libsystemd_la_LIBADD = \
 	libbasic.la \
 	libsystemd-journal-internal.la
 
-libsystemd-install-hook:
-	libname=libsystemd.so && $(move-to-rootlibdir)
-
-libsystemd-uninstall-hook:
-	rm -f $(DESTDIR)$(rootlibdir)/libsystemd.so*
-
-INSTALL_EXEC_HOOKS += libsystemd-install-hook
-UNINSTALL_EXEC_HOOKS += libsystemd-uninstall-hook
-
 pkgconfiglib_DATA += \
 	src/libsystemd/libsystemd.pc
 
@@ -3279,7 +3260,7 @@ pkginclude_HEADERS += \
 	src/systemd/sd-id128.h \
 	src/systemd/sd-daemon.h
 
-lib_LTLIBRARIES += \
+rootlib_LTLIBRARIES += \
 	libsystemd.la
 
 tests += \
@@ -3600,7 +3581,7 @@ tests += \
 include_HEADERS += \
 	src/libudev/libudev.h
 
-lib_LTLIBRARIES += \
+rootlib_LTLIBRARIES += \
 	libudev.la
 
 libudev_la_SOURCES =\
@@ -3632,16 +3613,6 @@ pkgconfiglib_DATA += \
 EXTRA_DIST += \
 	src/libudev/libudev.pc.in
 
-# move lib from $(libdir) to $(rootlibdir) and update devel link, if needed
-libudev-install-hook:
-	libname=libudev.so && $(move-to-rootlibdir)
-
-libudev-uninstall-hook:
-	rm -f $(DESTDIR)$(rootlibdir)/libudev.so*
-
-INSTALL_EXEC_HOOKS += libudev-install-hook
-UNINSTALL_EXEC_HOOKS += libudev-uninstall-hook
-
 # ------------------------------------------------------------------------------
 noinst_LTLIBRARIES += \
 	libudev-internal.la
